torch/nn/modules/transformer.py

```diff
@@ -15,6 +15,21 @@
 
 __all__ = ['Transformer', 'TransformerEncoder', 'TransformerDecoder', 'TransformerEncoderLayer', 'TransformerDecoderLayer']
 
+def _generate_square_subsequent_mask(
```

Comment:
Clarifying from the discussion offline, the reason for this helper is because we want to call this non-static variant in `_detect_is_causal_mask` so that `_detect_is_causal_mask` which is called by `TransformerEncoder.forward` / `TransformerDecoder.forward` can be torchscriptable is that correct?

Reply:
That is correct!